packages/twenty-front/src/modules/settings/developers/hooks/useWebhookForm.ts

```diff
@@ -0,0 +1,256 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useCreateOneRecord } from '@/object-record/hooks/useCreateOneRecord';
+import { useDeleteOneRecord } from '@/object-record/hooks/useDeleteOneRecord';
+import { useFindOneRecord } from '@/object-record/hooks/useFindOneRecord';
+import { useUpdateOneRecord } from '@/object-record/hooks/useUpdateOneRecord';
+import { WebhookFormMode } from '@/settings/developers/constants/WebhookFormMode';
+import { Webhook } from '@/settings/developers/types/webhook/Webhook';
+import {
+  webhookFormSchema,
+  WebhookFormValues,
+} from '@/settings/developers/validation-schemas/webhookFormSchema';
+import { SettingsPath } from '@/types/SettingsPath';
+import { SnackBarVariant } from '@/ui/feedback/snack-bar-manager/components/SnackBar';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { isDefined } from 'twenty-shared/utils';
+import { v4 } from 'uuid';
+import { useNavigateSettings } from '~/hooks/useNavigateSettings';
+import { WEBHOOK_EMPTY_OPERATION } from '~/pages/settings/developers/webhooks/constants/WebhookEmptyOperation';
+import { WebhookOperationType } from '~/pages/settings/developers/webhooks/types/WebhookOperationsType';
+
+type UseWebhookFormProps = {
+  webhookId?: string;
+  mode: WebhookFormMode;
+};
+
+export const useWebhookForm = ({ webhookId, mode }: UseWebhookFormProps) => {
+  const navigate = useNavigateSettings();
+  const { enqueueSnackBar } = useSnackBar();
+
+  const isCreationMode = mode === WebhookFormMode.Create;
+
+  const { createOneRecord } = useCreateOneRecord<Webhook>({
+    objectNameSingular: CoreObjectNameSingular.Webhook,
+  });
+
+  const { updateOneRecord } = useUpdateOneRecord<Webhook>({
+    objectNameSingular: CoreObjectNameSingular.Webhook,
+  });
+
+  const { deleteOneRecord: deleteOneWebhook } = useDeleteOneRecord({
+    objectNameSingular: CoreObjectNameSingular.Webhook,
+  });
+
+  const formConfig = useForm<WebhookFormValues>({
```

Comment:
Note that we are deprecating react hook form, as the precedent implementation was using it it's fine to keep it but let's not over invest in using it!